frontend/src/SearchView.svelte

```diff
@@ -0,0 +1,243 @@
+    height: 1em;
+    border: 1px solid currentColor;
+    display: grid;
+    place-content: center;
+    */
+  }
+
+  /*
+  input[type="checkbox"]:focus {
+    box-shadow: none;
+  }
+
+  input[type="checkbox"]::before {
+    content: "";
+    width: 0.45em;
+    height: 0.45em;
+    transform: scale(0);
+    transition: 120ms transform ease-in-out;
+    box-shadow: inset 1em 1em var(--main-fg-color);
+  }
+
+  input[type="checkbox"]:checked::before {
+    transform: scale(1);
+  }
+  */
+
+  label {
+    margin-bottom: 1em;
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    justify-content: space-evenly;
+    align-items: baseline;
+    align-content: center;
+    white-space: nowrap;
+  }
+
+  .row {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    justify-content: space-evenly;
+    align-items: baseline;
+    align-content: center;
+    white-space: nowrap;
+  }
+
+  .error {
+    margin-top: 2em;
+  }
+
+  .treebox {
+    flex-grow: 1;
+    padding-left: 1em;
+    overflow-x: scroll;
+    white-space: nowrap;
+    text-align: left;
+  }
+</style>
+
+<script>
+  import { selected, selectedResource } from "./stores.js";
+  import { calculator } from "./helpers";
+  import ResourceTreeNode from "./ResourceTreeNode.svelte";
+
+  export let modifierView, resourceNodeDataMap;
+  let searchInput,
+    searchRangeStartInput,
+    searchRangeEndInput,
+    rangeSearch = false,
+    results = [],
+    errorMessage;
+
+  const searchTarget = $selectedResource;
+
+  function refreshResource() {
+    // Force hex view refresh with colors
+    const originalSelected = $selected;
+    $selected = undefined;
+    $selected = originalSelected;
+  }
+
+  async function search() {
+    if (searchTarget) {
+      try {
+        if (rangeSearch) {
+          const searchRangeStartAddress = calculator.calculate(
+              searchRangeStartInput
+            ),
+            searchRangeEndAddress = calculator.calculate(searchRangeEndInput);
+
+          results = await searchTarget.search_for_vaddr(
+            searchRangeStartAddress,
+            searchRangeEndAddress
+          );
+        } else {
+          const startAddress = calculator.calculate(searchInput);
+
+          results = await searchTarget.search_for_vaddr(startAddress, null);
+        }
+      } catch (err) {
+        try {
+          errorMessage = JSON.parse(err.message).message;
+        } catch (_) {
+          errorMessage = err.message;
+        }
+      }
+    }
+  }
+  async function leave_view() {
+    if ($selectedResource !== undefined) {
+      const ancestors = await $selectedResource.get_ancestors(null);
+      for (const ancestor of ancestors) {
+        resourceNodeDataMap[ancestor.get_id()].collapsed = false;
+      }
+    }
+    modifierView = undefined;
+  }
+</script>
+
+<div class="container">
+  <div class="inputs">
+    <p>
+      Searching for descendants of {searchTarget.get_id()} whose virtual address
```

Comment:
It may be more helpful to have the resource's label instead of its ID here? Perhaps both?

Reply:
I specifically chose not to use the label because it isn't unique to the resource. I wanted it to be clear that the same resource is being searched. I don't think having the label here is helpful as the user should already know what resource they are looking at, having selected it in the main tree view.

Reply:
I am going to push back on this a little. If the user misclicks a resource or has one highlighted that they didn't intend, then this doesn't actually provide them with any useful information. While it's true that they could look at the attributes for the highlighted resource, I think having the full hex of the UUID front-and-center is not actually helpful.

If the resource ID is necessary, I think this should have both the resource ID and the label.

Reply:
I can add the label, but arguably the root of the problem is that the GUI does not display any unique identifier for resources at the moment. A lot of the labels overlap ("ElfSection", "ElfSection, CodeRegion", "File" etc.) and the UUID is not displayed anywhere.

Reply:
I think displaying the UUID in the attributes view area would solve both of those problems. Then it's showing whenever you click a resource, and when it's selected to modify. 

I would argue that for regular browsing around, having the UUID display for every resource is overstimulating and mostly uninformative. It's only useful when you want to confirm the selected resource has the correct ID. Otherwise, the consistent order of the child resources allows for a basic level of differentiation between resources with the same labels.